meshtastic/mesh_interface.py

```diff
@@ -661,22 +661,32 @@ def onResponseTelemetry(self, p: dict):
             self._acknowledgment.receivedTelemetry = True
             telemetry = telemetry_pb2.Telemetry()
             telemetry.ParseFromString(p["decoded"]["payload"])
-
             print("Telemetry received:")
-            if telemetry.device_metrics.battery_level is not None:
-                print(f"Battery level: {telemetry.device_metrics.battery_level:.2f}%")
-            if telemetry.device_metrics.voltage is not None:
-                print(f"Voltage: {telemetry.device_metrics.voltage:.2f} V")
-            if telemetry.device_metrics.channel_utilization is not None:
-                print(
-                    f"Total channel utilization: {telemetry.device_metrics.channel_utilization:.2f}%"
-                )
-            if telemetry.device_metrics.air_util_tx is not None:
-                print(
-                    f"Transmit air utilization: {telemetry.device_metrics.air_util_tx:.2f}%"
-                )
-            if telemetry.device_metrics.uptime_seconds is not None:
-                print(f"Uptime: {telemetry.device_metrics.uptime_seconds} s")
+            # Check if the telemetry message has the device_metrics field
+            # This is the original code that was the default for --request-telemetry and is kept for compatibility
+            if telemetry.HasField("device_metrics"):
+                if telemetry.device_metrics.battery_level is not None:
+                    print(f"Battery level: {telemetry.device_metrics.battery_level:.2f}%")
+                if telemetry.device_metrics.voltage is not None:
+                    print(f"Voltage: {telemetry.device_metrics.voltage:.2f} V")
+                if telemetry.device_metrics.channel_utilization is not None:
+                    print(
+                        f"Total channel utilization: {telemetry.device_metrics.channel_utilization:.2f}%"
+                    )
+                if telemetry.device_metrics.air_util_tx is not None:
+                    print(
+                        f"Transmit air utilization: {telemetry.device_metrics.air_util_tx:.2f}%"
+                    )
+                if telemetry.device_metrics.uptime_seconds is not None:
+                    print(f"Uptime: {telemetry.device_metrics.uptime_seconds} s")
+            else:
+                # this is the new code if --request-telemetry <type> is used.
```

Comment:
Probably need to add a guard for some sort of 'received empty or garbled TELEMETRY_APP packet' here. Otherwise we'll choke on this. The old version at least checked for `telemetry.HasField("device_metrics")`

Reply:
Since we're parsing the protobuf out of the payload up above on line 662, I think it wouldn't get here at all if it's garbled in some fashion we could usefully detect. If it got an empty value back, the MessageToDict wouldn't have keys in it anyway, so it'd just print "Telemetry received:" and then nothing. Which, while a little confusing, probably not a big deal. I guess maybe it'd be reasonable to check what happens when requesting telemetry types not supported by a given node, since that'd possibly (probably?) return a payload with the right `oneof` selected but nothing to put into it. But I think it'd still print something comprehensible enough, so not too worried about it.